--- ../src-base/minecraft/net/minecraft/world/end/DragonFightManager.java
+++ ../src-work/minecraft/net/minecraft/world/end/DragonFightManager.java
@@ -25,6 +25,7 @@
 import net.minecraft.entity.item.EntityEnderCrystal;
 import net.minecraft.entity.player.EntityPlayerMP;
 import net.minecraft.init.Blocks;
+import net.minecraft.init.Items;
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.nbt.NBTTagInt;
 import net.minecraft.nbt.NBTTagList;
@@ -45,9 +46,15 @@
 import net.minecraft.world.gen.feature.WorldGenEndGateway;
 import net.minecraft.world.gen.feature.WorldGenEndPodium;
 import net.minecraft.world.gen.feature.WorldGenSpikes;
+
+import redstone.multimeter.common.TickTask;
+import redstone.multimeter.helper.WorldHelper;
+
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
+import carpet.CarpetSettings;
+
 public class DragonFightManager
 {
     private static final Logger field_186107_a = LogManager.getLogger();
@@ -149,6 +156,8 @@
 
     public void func_186105_b()
     {
+        WorldHelper.startTickTask(TickTask.DRAGON_FIGHT); // RSMM
+
         this.field_186109_c.func_186758_d(!this.field_186117_k);
 
         if (++this.field_186116_j >= 20)
@@ -244,6 +253,8 @@
                 }
             }
         }
+
+        WorldHelper.endTickTask(); // RSMM
     }
 
     protected void func_186095_a(DragonSpawnManager p_186095_1_)
@@ -405,6 +416,10 @@
 
             this.field_186118_l = true;
             this.field_186117_k = true;
+
+            if (CarpetSettings.renewableElytras) {
+                p_186096_1_.func_145779_a(Items.field_185160_cR, 1);
+            }
         }
     }
 
@@ -437,6 +452,8 @@
             }
         }
 
+        // Fix for the end portal somehow spawning at y = -2 when spawning the first time in skyblock CARPET-XCOM
+        if(CarpetSettings.skyblock && field_186121_o.func_177956_o() <= 0) field_186121_o = new BlockPos(field_186121_o.func_177958_n(), 63, field_186121_o.func_177952_p());
         worldgenendpodium.func_180709_b(this.field_186110_d, new Random(), this.field_186121_o);
     }
 
